engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/date/EnsoDate.java

```diff
@@ -0,0 +1,110 @@
+package org.enso.interpreter.node.expression.builtin.date;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.TruffleObject;
+import com.oracle.truffle.api.interop.UnknownIdentifierException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import org.enso.interpreter.runtime.Context;
+import org.enso.interpreter.runtime.callable.UnresolvedSymbol;
+import org.enso.interpreter.runtime.callable.function.Function;
+import org.enso.interpreter.runtime.data.Array;
+import org.enso.interpreter.runtime.library.dispatch.MethodDispatchLibrary;
+
+@ExportLibrary(InteropLibrary.class)
+@ExportLibrary(MethodDispatchLibrary.class)
+public final class EnsoDate implements TruffleObject {
+  private final LocalDate date;
+
+  public EnsoDate(LocalDate date) {
+    this.date = date;
+  }
+
+  @ExportMessage
+  boolean isDate() {
+    return true;
+  }
+
+  @ExportMessage
+  LocalDate asDate() {
+    return date;
+  }
+
+  @ExportMessage
+  boolean hasMembers() {
+    return true;
+  }
+
+  @ExportMessage
+  Object getMembers(boolean internal) {
+    return new Array("year", "month", "day");
+  }
+
+  @ExportMessage
+  boolean isMemberReadable(String member) {
+    return switch (member) {
+      case "year" -> true;
+      case "month" -> true;
+      case "day" -> true;
+      default -> false;
+    };
+  }
+
+  @ExportMessage
+  int readMember(String member, @Cached("member") String cachedMember) throws UnknownIdentifierException {
```

Comment:
Define the proper methods instead: can be done through Hubert's DSL – see `EnsoFile`

Reply:
Done in [c98723c](https://github.com/enso-org/enso/pull/3559/commits/c98723cceb86ce714e64d5e9790e7818c09c854e)